### ISSUE TITLE
AuditedEntityWithUserDto when generating proxy for angular

### ISSUE THREAD

**Issue by wael-razouk** [abpframework/abp#17142]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

I have an app service called `CategoryAppService` as follows:
```public class CategoryAppService : ICategoryAppService
    {
        public CategoryAppService() { }

        public List<CategoryDto> GetAll()
        {
            return new List<CategoryDto>();
        }
    }
```
and here is the `CategoryDto` class
```
public class CategoryDto : AuditedEntityWithUserDto<Guid, IdentityUserDto>
    {
        public string Name { get; set; }
    }
```
When I generate the proxies for the angular project using command `abp generate-proxy -t ng`
I get this `models.ts` file
```
import type { AuditedEntityWithUserDto } from '@abp/ng.core';

export interface CategoryDto extends AuditedEntityWithUserDto<string, IdentityUserDto> {
  name?: string;
}
```
Here we have 2 problems:
- `IdentityUserDto` is not defined (missing adding import for it from @abp/ng.identity/proxy)
- `AuditedEntityWithUserDto` in angular project is defined as `AuditedEntityWithUserDto<TUserDto, TPrimaryKey>` (which is the opposite in the backend project (`TUserDto` and `TPrimaryKey` are replaced in position), but here in this `models.ts` they are wrongly placed. So now this model has its primary key of type `IdentityUserDto` and `Creator`, `LastModifier`, `Deleter` are of type `string`

P.S: This issue exists for any audited entity with `UserDto` (i.e `AuditedEntityWithUserDto`, `CreationAuditedEntityWithUserDto`, `FullAuditedEntityWithUserDto`, ...)

### Version

7.2

### User Interface

Angular

### Database Provider

EF Core (Default)

### Tiered or separate authentication server

None (Default)

### Operation System

Windows (Default)

**Comment by wael-razouk** [comment]
@maliming  new issue as suggested

**Comment by wael-razouk** [comment]
@maliming 

Specifying the core problem:
The issue is not with `IdentityUserDto` itself, rather than it appears when we inherit a class with non-primitive generic parameter.

Example:
If I define the class `CategoryDto` as follows:
```
namespace Comp.Tutorial.Categories.Dtos
{
    public class CategoryDto : GenericDto<GenericParam>
    {
      
    }
}
```
Here `GenericDto` and `GenericParam` are classes from another namespace. When generating proxies, I get for `categories/dtos/models.ts` file:
```
import type { GenericDto } from '../../generics/dtos/models';

export interface CategoryDto extends GenericDto<GenericParam> {
}

```
Notice here the import for `GenericDto` only.

**Comment by wael-razouk** [comment]
@maliming  any plans to fix this one?

**Comment by wael-razouk** [comment]
@mahmut-gundogdu  Any ongoing work on this?

**Comment by mahmut-gundogdu** [comment]
> * AuditedEntityWithUserDto

I have produced. I am working on it.

**Comment by mahmut-gundogdu** [comment]
> @mahmut-gundogdu Any ongoing work on this?

I change order of AuditedEntityWithUserDto<primaryKey,UserDto> in typescript but It will release in V8.0. It cause a breaking change so We can't fix in 7.2 or newer. So my suggestion is you should remove the inheritance, add the properties manually. Then it will fixed.

Missing IdentityUserDto import will fixed in patch version.
